Assets/Code/Scripts/BulletPool.cs

```diff
@@ -0,0 +1,77 @@
+using UnityEngine;
+using System.Collections.Generic;
+
+
+namespace MJ.Weapon
+{ 
+public class BulletPool : MonoBehaviour
+{
+    public static BulletPool Instance;
+
+    // 각 무기 ID에 따른 총알 프리팹을 저장하는 Dictionary
+    public Dictionary<int, GameObject> bulletPrefabs = new Dictionary<int, GameObject>();
+
+    // 각 무기 ID에 따른 총알 풀을 저장하는 Dictionary
+    private Dictionary<int, Queue<GameObject>> bulletPools = new Dictionary<int, Queue<GameObject>>();
+
+    private void Awake()
+    {
+        if (Instance == null)
+        {
+            Instance = this;
```

Comment:
```suggestion
            Instance = this;
            DontDestroyOnLoad(gameObject);
```

불릿 풀이 의도되지 않게 여러개 생성될 수 있으니, DontDestroyOnLoad를 추가하는 것이 좋을 것 같습니다